submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/SubmissionApiTest.kt

```diff
@@ -159,6 +161,31 @@ internal class SubmissionApiTest(private val tempFolder: TemporaryFolder) : Base
             )
         }
 
+        @Test
+        fun `submission with on behalf`() {
+            createUser(RegularUser, serverPort)
+
+            val submission = tsv {
+                line("Submission")
+                line("Title", "Submission Title")
+            }.toString()
+
+            val onBehalfClient = SecurityWebClient
+                .create("http://localhost:$serverPort")
+                .getAuthenticatedClient(SuperUser.email, SuperUser.password, RegularUser.email)
+
+            val response = onBehalfClient.submitSingle(submission, SubmissionFormat.TSV)
+            assertThat(response).isSuccessful()
+
+            val accNo = response.body.accNo
+            assertThat(submissionRepository.getByAccNo(accNo)).isEqualTo(
+                submission(accNo) {
+                    title = "Submission Title"
+                    releaseDate = LocalDate.now().toString()
+                }
+            )
+        }
+
         @Test
```

Comment:
It would be good to have a negative case where a superuser impersonates a regular user and try to set a specific accession, which shouldn't be allowed.